### ISSUE TITLE
[FEATURE]: Assign hotkeys

### ISSUE THREAD

**Issue by DegoDani** [MissCorruption/QuickLootIE#51]
### Feature Title

Assign hotkeys

### Motivation

The default L3 to open loot menu clashes with other bindings. This is probably only an issue due to other inputs not being suppressed properly, but the option to assign keys would be very beneficial.

### Feature Description

Mcm submenu (or .ini) wherein a player can bind quickloot menu usage to keys of choice.
Additionally, some form of autoload of settings on new game would be very satisfying.

**Comment by AtomCrafty** [comment]
This is planned, but it's going to take a while to implement.

**Comment by MartinHaeusler** [comment]
As a workaround for now I would also be happy with an INI file where I can configure the key codes. I really love what Quickloot IE does, but sadly it's nearly unusable with a controller at the moment because the d-pad up key opens the container, which makes navigation through the menu one-way only.

**Comment by AtomCrafty** [comment]
The infrastructure for this feature is done as of #64. What's missing is the MCM integration and some code to show the correct keybindings in the button bar of the loot menu. Our resident MCM expert is busy at the moment, so I'll have to figure that out myself somehow. Until that is done you can bind the favorites menu to a different key, since that is the keybinding the current release version uses for the "Transfer" button.

**Comment by MartinHaeusler** [comment]
Thank you for your efforts, higly appreciated! I will give the favorite menu rebinding a try.

**Comment by AtomCrafty** [comment]
I'm not entirely sure I understand that. Does Skyrim even allow binding actions to key combinations like <kbd>A</kbd>+<kbd>D_UP</kbd>? I don't play with controller myself, so I'm not very familiar with the gamepad settings.

Are you saying that binding <kbd>D_UP</kbd> to any action other than favorites breaks _all_ keybindings globally? Or just keybindings that also include <kbd>D_UP</kbd>? Is that related to QuickLoot or just a Skyrim thing in general?

**Comment by MartinHaeusler** [comment]
@AtomCrafty you can assign key combinations in vanilla skyrim in the `controlmap.txt`. This is also shown in tools like [Skyrim Control Mapper](https://hawk.bar/SkyrimControlMapper/). This is basically all just configuration for what vanilla skyrim can already do (but doesn't offer a native UI for it, god forbid):

![image](https://github.com/user-attachments/assets/43934992-c07c-4468-af26-637ed66f2ee8)

In the screenshot above, you'd need to hold down the A button on the controller and then you could use the d-pad to perform various actions (opening the 4 main menu screens in this case), which is very useful because the number of buttons on a gamepad is obviously very limited compared to a full keyboard.

What I discovered to day is that if you **unmap** the functionality on the basic D-pad up/down (without any other modifier, just these two) to make room for a mod (e.g. quickloot) then all the combinations (e.g. A+d-pad up) simply stop working and won't do anything. I don't know yet if this is specific to the favorites menu or not.

**TL;DR:**

Skyrim's input system is bugged (surprise!) and it's hard to completely un-map the inputs for the d-pad. This would make configurable inputs for Quickloot IE all the more valuable, in particular for gamepad users. Quickloot IE already "grabs" the d-pad inputs when the popup is open and prevents them from triggering something else, e.g. the favorites menu (which is great!). All that's missing from my point of view is the ability to remap the "open container" functionality to some other button so we can navigate through the quickloot menu on the controller in both directions.

**Comment by AtomCrafty** [comment]
Okay, I see. The A button is the modifier key for those keybindings. I'm painfully familiar with how the control map works due to the recent rewrite of QLIE's input system.

Either way, that doesn't really seem relevant to QLIE, but an interesting observation nonetheless. On a different note, I'd be interested what you would consider an appropriate default key for the Transfer action. I just bound it to Y for testing, but it might be annoying to be unable to jump while looking at a body. The keybindings will be configurable, but it would still be best if we provided sensible defaults.

https://github.com/MissCorruption/QuickLootIE/blob/c4de9f5977cf6aacc4337a1b0d0c526e5246e5ce/src/Input/InputManager.cpp#L217-L224

**Comment by MartinHaeusler** [comment]
I think that opening the full-blown container menu is a rather disruptive action and usually something the user will want to avoid doing by accident (that's kind of the point of the Quickloot idea after all). So I would not map it to one of the four main face buttons. Maybe the "back" button on the 360 gamepad? It's far enough away from the normal grip to be sure that the action is intentional.

**Comment by AtomCrafty** [comment]
MCM is done and hooked up. This can be closed now.

**Comment by MartinHaeusler** [comment]
Great, thank you! Looking forward to the new version on the nexus :)
